src/Chess/ChessBoard.hs

```diff
@@ -31,7 +31,28 @@ chessBoardRowAsString rowNum chessBoard = rowNumStr ++ "  " ++ (foldl (\ rowStr
 -- WARNING: Assumes given move is valid!
 -- TODO: implement + test this (1.5 hours) [Yiyi]
 makeMove :: ChessBoard -> ChessMove -> ChessBoard
-makeMove chessBoard move = chessBoard
+makeMove cb@(ChessBoard pieces state) (ChessMove from to) = 
+    ChessBoard (putPiece pieceToMove to (removePiece from pieces)) (if (isCheckMate cb (getPieceColour pieceToMove)) then Over else state)
```

Comment:
I think you need to do both `removePiece from pieces` and `removePiece to pieces` because there can be a piece at the `to` position as well (in case the pieceToMove is killing another piece).